### DISCUSSION TITLE
pausing a queue?

### DISCUSSION THREAD

**Discussion by ryanking** [Mergifyio/mergify#3984] [Q&A] [answered]
Sorry if this is a dumb question (I am not a user yet, but trying to figure some things out).

Is it possible to pause a queue? Or only allow high-priority items through?

Let me explain - sometimes when we are in incident management or other operational states we block all merges other than ones addressing the crisis. Is there a way to set mergify to (temporarily) only merge PRs over a certain priority level?

**Comment by jd** [comment]
Hey @ryanking!

For now, you can pause a queue by disabling the rules, which means editing the configuration file. This is not great as it would cancel running builds. Unfortunately, Mergify does not offer a toggle yet to pause/unpause a queue, but this is on our roadmap and should be available soon.

In the meantime, a solution would be to use a check (e.g., a GitHub Action) that returns failure if your queue is meant to be paused. This is far from perfect, as it'll empty the queue and will cancel all the build — but it's a workaround.

**Reply by ryanking** [reply]
I assume that mergify uses the config file on the default branch?

**Reply by jd** [reply]
Yes, you are right on this.

**Comment by sileht** [comment, marked as answer]
We now offer an API to do that: https://docs.mergify.com/api/#/operations/create_queue_freeze_repos__owner___repository__queue__queue_name__freeze_put
